### DISCUSSION TITLE
How can I set GotFocus/LostFocus events to every TextBox?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#12124] [Q&A] [answered]
Recently I have been coding a Virtual Keyboard with IME for DRM in Linux.

I need to show the Virtual Keyboard when the TextBox GotFocus and hide it when TextBox LostFocus.

However, currently I need to add GotFocus/LostFocus events to every TextBox one by one.

It is so troublesome.

Is there any way that I can add events more conveniently?

**Comment by timunie** [comment, marked as answer]
Use Avalonia.Behaviors and write your own Behavior for it. This can be attached via Style

**Reply by kyuranger** [reply]
Thank you. What's more, how can I add a style with the Behavior by code? I want the Virtual Keyboard only show in Linux DRM and not show in any other platform.

**Reply by timunie** [reply]
please explore the samples apps they provide. Also I'd just apply it on every OS but just skip the behavior on not DRM environment. Or use a style class

**Reply by kyuranger** [reply]
Thanks a lot. Let me try it.

**Comment by maxkatz6** [comment]
You can subscribe on `InputElement.GotFocusEvent.AddClassHandler`.

**Reply by kyuranger** [reply]
Thanks a lot! This way seems more convenient!
